### PR TITLE
Refactor view store non singleton

### PR DESCRIPTION
**Pull request by bglusman**
@yaymukund may conflict with your template store refactor but should be pretty good otherwise.

### CODE REVIEW

File: lib/view_store.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-exports.getView = function(name) {
-  if (VIEWS[name]) {
-    return VIEWS[name];
+  this.load = function(name, View) {
+    VIEWS[name] = View;
+  };
 
-  } else if (typeof getView === 'function') {
-    return getView(name);
+  this.getView = function(name) {
```
Comment (yaymukund) [inline lib/view_store.js]: You're declaring functions on the instance directly. This means that we'll create a copy of `load`, `getView`, `setCustomGetView` every time we call `new ViewStore()`. Instead, you can declare functions on the prototype:

``` js
function ViewStore() {
  this.VIEWS = {};
  this.getView = null;
}

ViewStore.prototype.load = function(name, View) {
  this.VIEWS[name] = View;
};

ViewStore.prototype.getView = function(name) { ... }
```

Comment (bglusman) [inline lib/view_store.js]: Ahh, right, forgot, thanks...

Comment (bglusman) [inline lib/view_store.js]: Curious about the this.getView = null though... kind of follow what with setCustomGetView, but little fuzzy why need explicit null, won't it be null on instance by default?

Anyway, so far get 3 errors in tests when I set that to null, and one without it, as if VIEWS were undefined, but thx for review/help.

**Comment by bglusman** [comment]
@yaymukund merged in your templateStore refactor because they were interdependent/doing much of the same stuff...  @devmanhinton thanks for your help, tests now pass, hopefully this is ready for CR and merge now.

File: lib/reactions/view.js [hunk 2]
```diff
@@ -3,12 +3,12 @@ var Reaction = require("../reaction"),
...
     _ = require("underscore"),
-    ViewStore = require('../view_store');
+    EndDash = require('../end-dash');
 
 var ViewReaction = Reaction.extend({
   init: function(next) {
     var oldView = this.el.data("view"),
-        viewClass = ViewStore.getView(this.viewName),
+        viewClass = EndDash.viewStore.getView(this.viewName),
         view, opts;
```
Comment (devmanhinton) [inline lib/reactions/view.js:12]: We need to decide wether to do:

``` javascript
EndDash.viewStore.getView(...
```

Or define a wrapper on EndDash and do:

``` javascript
EndDash.getView(this.viewName..
```

Just need to pick one and stay consistent. This is an open question for you, Mukund, and me.

Comment (yaymukund) [inline lib/reactions/view.js:12]: We should do `EndDash.viewStore.getView(...)` within EndDash. There might eventually be `viewStore` methods that we'll want to use internally, but we don't want to expose on `EndDash`.

**Comment by devmanhinton** [comment]
@bglusman  This seems good to me. Mukund's branch is merged into this one. That branch is about to be updated with some changes. 

Maybe:

1) Mukund fixes his branch, pushes -> we CR & merge
2) Make things mergeable
      --- a) View_store
      --- b) New-collection-iteration
3) Meanwhile Mukund can CR
4) Back & forth code review/edits on 2a & 2b
5) Merge 2a & 2b

Then we have:
1) New collection iteration
2) More prototypal and non-singleton modules

Finally:

1) PR new-collection-iteration-deprecation (Mukund CR, we refactor based on comments)
2) Refactor the parser/template_store/end-dash

Hopefully all by EOD

File: lib/reactions/view.js (hunk 2 above)
Comment (devmanhinton) [inline lib/reactions/view.js:12]: Sure, leaving it this way for now

**Pull request merged**